### DISCUSSION TITLE
How to customize generated connection types?

### DISCUSSION THREAD

**Discussion by devopsix** [graphql-java-kickstart/graphql-java-tools#470] [Q&A] [answered]
I am wondering what the best way for customizing the connection types generated for `@connection` directives would be.

Let's say I want to implement a GraphQL service matching [this example from the Introduction to GraphQL](https://graphql.org/learn/queries/#using-variables-inside-fragments) which has an extra `totalCount` field on the connection:

```graphql
query HeroComparison($first: Int = 3) {
  leftComparison: hero(episode: EMPIRE) {
    ...comparisonFields
  }
  rightComparison: hero(episode: JEDI) {
    ...comparisonFields
  }
}

fragment comparisonFields on Character {
  name
  friendsConnection(first: $first) {
    totalCount
    edges {
      node {
        name
      }
    }
  }
}
```

As far as I understand [`RelayConnectionFactory.kt`](https://github.com/graphql-java-kickstart/graphql-java-tools/blob/6.3.0/src/main/kotlin/graphql/kickstart/tools/relay/RelayConnectionFactory.kt), I would have to replace it entirely for adding any extra field.

Is there a smarter way than replacing `RelayConnectionFactory`?

**Comment by vojtapol** [comment, marked as answer]
Currently, the only way to achieve that is to indeed write your own type factory. You can start by copy pasting the existing `RelayConnectionFactory`. Moreover, you will need to choose a different directive as `@connection` is already taken by the existing Relay implementation.
